src/main/java/com/project/bookstore/repository/book/spec/AuthorSpecificationProvider.java

```diff
@@ -17,8 +17,7 @@ public String getKey() {
 
     @Override
     public Specification<Book> getSpecification(String[] params) {
-        return (((root, query, criteriaBuilder) -> root
-                .get(KEY).in(Arrays.stream(params)
-                        .toArray())));
+        return ((root, query, criteriaBuilder) -> root
+                .get(KEY).in(Arrays.stream(params).toArray()));

```

Comment:
Why do we need this action? you doing any transformations in stream. 
```suggestion
                .get(KEY).in(params);

```